ansible_base/authentication/utils/authentication.py

```diff
@@ -40,19 +92,28 @@ def check_system_username(uid: str) -> None:
         raise AuthException(_('System user is not allowed to log in from external authentication sources.'))
 
 
-def determine_username_from_uid_social(**kwargs: dict) -> dict:
-    uid = kwargs.get('details', {}).get('username', None)
-    if not uid:
+def determine_username_from_uid_social(**kwargs) -> dict:
```

Comment:
Continuing from another comments, I would push for a different name for this method, because it does a whole lot more than what it looks like it does.

Reply:
I don't really want to mess with the name of this function, because it gets included in the SOCIAL_AUTH_PIPELINE setting. I know we configure this in the dynamic settings, but it would break any app that isn't using dynamic settings.

I could probably split this into a separate pipeline stage, but I know that this works right now from manual testing. We can't test the full pipeline in CI right now, so I'm not confident enough that I can make this change without breaking something else.